hq/app/schedule/IamListAccessKeys.scala

```diff
@@ -0,0 +1,48 @@
+package schedule
+
+import aws.AwsAsyncHandler.{awsToScala, handleAWSErrs}
+import aws.iam.IAMClient.SOLE_REGION
+import aws.{AwsClient, AwsClients}
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementAsync
+import com.amazonaws.services.identitymanagement.model.{AccessKeyMetadata, ListAccessKeysRequest, ListAccessKeysResult}
+import model.{AccessKeyWithId, AwsAccount, VulnerableUser, VulnerableUserWithAccessKeyId}
+import utils.attempt.Attempt
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+object IamListAccessKeys {
+
+  // get information on all access keys for a given AWS account
+  def listAccountAccessKeys(account: AwsAccount, users: Seq[VulnerableUser], iamClients: AwsClients[AmazonIdentityManagementAsync])
+    (implicit ec: ExecutionContext): Attempt[List[VulnerableUserWithAccessKeyId]] = {
+    val accessKeyData = users.map { user =>
+      for {
+        client <- iamClients.get(account, SOLE_REGION)
+        keys <- listAccessKeys(client, user)
+      } yield keys.getAccessKeyMetadata
+    }.map(attempt => attempt.map(metadatas => metadatas.asScala.toList)).toList
+
+    Attempt.sequence(accessKeyData).map(_.flatten).map(accessKeyMetadatas => addAccessKeysToUsers(accessKeyMetadatas, users))
```

Comment:
Also - we could add a new method to `Attempt` called `flatSequence` that does `Attempt.sequence(accessKeyData).map(_.flatten)` as I see there's already a `flatTraverse`. It could be implemented in terms of `flatTraverse` even!